src/sage/modular/quasimodform/element.py

```diff
@@ -728,3 +730,73 @@ def derivative(self):
         hom_comp = self.homogeneous_components()
 
         return sum(f.serre_derivative() + R(k) * u * f * E2 for k, f in hom_comp.items())
+
+    def _compute(self, X):
+        r"""
+        Compute the coefficients of `q^n` of the q-expansion of this,
+        graded quasimodular form for `n` in the list `X`. 
+        
+        The results are not cached.  (Use coefficients for cached results).
+
+        EXAMPLES::
+
+            sage: E2 = QuasiModularForms(1).0
+            sage: E2.q_expansion(10)
+            1 - 24*q - 72*q^2 - 96*q^3 - 168*q^4 - 144*q^5 - 288*q^6 - 192*q^7 - 360*q^8 - 312*q^9 + O(q^10)
+            sage: E2._compute([3, 6])
+            [-96, -288]
+            sage: E2._compute([])
+            []
+        """
+        if not isinstance(X, list) or not X:
+            return []
+        bound = max(X)
+        q_exp = self.q_expansion(bound + 1)
+        return [q_exp[i] for i in X]        
+
+    def coefficients(self, X):
+        r"""
+        Return the coefficients of `q^n` of the q-expansion of this,
+        graded quasimodular form for `n` in the list `X`.
+        
+        If X is an Integer, return coefficients for indices from 1
+        to X. This function caches the results of the `_compute` function.
+
+        TESTS::
```

Comment:
```suggestion
        EXAMPLES::
```
If you use the "TESTS" block, the doctests will not appear in the reference manual.

Reply:
But it would be still tested, right? (is `TESTS::` legacy?)

Reply:
Yes both `TESTS` and `EXAMPLES` run the doctests, however `TESTS` block does not show any examples in the Sage reference manual and I think that it is the only difference. I personnaly think that it is a good thing to present some examples in the manual so the user can better understand how to use the method.

Reply:
To add to my previous reply, even though it is a good practice to present some examples, sometimes some doctests aren't that useful for the user (technical or internal doctest) so you might want to hide them. But in your case the doctests are all relevant so that is why I suggested to use the `EXAMPLES` block.

Reply:
Thanks, I didn't know that examples would be also tested. Just updated!